components/tikv_util/src/codec/stream_event.rs

```diff
@@ -16,53 +16,48 @@ pub trait Iterator {
 
 pub struct EventIterator {
     buf: Cursor<Vec<u8>>,
-    index: usize,
```

Comment:
i think you should keep the `index` and not use `Cursor<...>`

Reply:
I'm not sure why... If we keep the `Cursor`, we can keep using `read_le_u32` (instead of implement it ourselves) and IMHO, `Cursor` is exactly `(buf, index)`.🤔

Reply:
you could use `byteorder::LittleEndian::read_u32(&buf[index..])` 🤔